src/App.tsx

```diff
@@ -1,14 +1,75 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
+import { getTodos } from './api';
+
+import { Todo } from './types/Todo';
+import { Filter } from './types/Filter';
+
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [filter, setFilter] = useState(Filter.ALL);
+  const [query, setQuery] = useState('');
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    getTodos()
+      .then((todosFromServer) => {
+        setTodos(todosFromServer);
+      })
+      .catch((error) => {
+        throw new Error(error.message);
+      })
+      .finally(() => setIsLoading(false));
+  }, []);
+
+  const visibleTodos = useMemo(() => {
+    return todos.filter((todo) => {
+      const isTodoSuitable = todo.title
+        .toLowerCase()
+        .trim()
+        .includes(query.toLowerCase().trim());
+
+      switch (filter) {
+        case Filter.ALL:
+          return isTodoSuitable;
+
+        case Filter.ACTIVE:
+          return isTodoSuitable && !todo.completed;
+
+        case Filter.COMPLETED:
+          return isTodoSuitable && todo.completed;
+
+        default:
+          throw new Error('Unknown filter');
+      }
+    });
```

Comment:
No need to filter in case when Filter is All and query is empty

Reply:
Ok, I think I got it. Can you, please, look at my fix below